src/zenml/integrations/kubernetes/service_connectors/kubernetes_service_connector.py

```diff
@@ -255,29 +255,44 @@ def _connect_to_resource(
                     client_cert.encode("utf-8")
                 )
 
-                with tempfile.NamedTemporaryFile(delete=False) as fp:
+                fd, temp_path = tempfile.mkstemp()
+                fp = os.fdopen(fd, "wb")
+                try:
                     fp.write(client_cert_bs)
-                    k8s_conf.cert_file = fp.name
+                    fp.flush()
+                finally:
+                    fp.close()
+                k8s_conf.cert_file = temp_path
```

Comment:
I see this logic duplicated several times below, should we move this to a separate util function perhaps? TBH I also don't fully understand what it does so having this in one place with corresponding docstrings would be helpful as well.